actions/actions.py

```diff
@@ -183,6 +192,72 @@ def validate_province(
     ) -> Dict[Text, Optional[Text]]:
         return self.validate_generic("province", dispatcher, value, self.province_data)
 
+    def validate_location(
+        self,
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Dict[Text, Optional[Text]]:
+        if not value:
+            dispatcher.utter_message(template="utter_incorrect_selection")
+            return {"location": None}
+
+        if not config.GOOGLE_PLACES_API_KEY:
+            return {
+                "location": value,
+                "latitude": "null",
+                "longitude": "null",
+            }
+
+        querystring = urlencode(
+            {
+                "key": config.GOOGLE_PLACES_API_KEY,
+                "input": value,
+                "language": "en",
+                "inputtype": "textquery",
+                "fields": "formatted_address,geometry",
+            }
+        )
+        response = requests.get(
+            (
+                f"https://maps.googleapis.com"
+                f"/maps/api/place/findplacefromtext/json?{querystring}"
+            )
+        )
```

Comment:
`rasa-sdk` runs as a Sanic web application, which is async, so I don't think that we should be doing blocking network requests inside of it. I used `httpx` in the connector to make async requests, it has an API similar to `requests`: https://github.com/praekeltfoundation/turn-rasa-connector/blob/develop/turn_rasa_connector/turn.py#L55-L65 . Just be careful with looking up docs. The version included with `rasa` (and `rasa-sdk`) is quite old, `0.9.3`, and has a completely different API than if you look up the docs for `httpx`

Reply:
I'm going to look at `httpx` but they do use `requests` in their demo https://github.com/RasaHQ/rasa-demo/blob/master/actions/actions.py#L772